src/CoreMidi/MidiThruConnectionParams.cs

```diff
@@ -327,19 +327,19 @@ internal NSData WriteStruct ()
 				Marshal.StructureToPtr (connectionParams, buffer, false);
 
 				if (connectionParams.NumControlTransforms > 0) {
-					unsafe { // Lets use memcpy to avoid a for loop
-						fixed (MidiControlTransform* arrAddr = &Controls[0])
-							Runtime.memcpy (bufferEnd, (IntPtr) arrAddr, controlsSize * connectionParams.NumControlTransforms);
+					unsafe {
+						fixed (void* arrAddr = &Controls[0])
+							Buffer.MemoryCopy ((void*) bufferEnd, arrAddr, controlsSize * connectionParams.NumControlTransforms, controlsSize * connectionParams.NumControlTransforms);
 					}
 				}
 
 				if (connectionParams.NumMaps > 0) {
 					// Set the destination buffer after controls arr if any
 					bufferEnd = IntPtr.Add (bufferEnd, controlsSize * connectionParams.NumControlTransforms);
-					unsafe { // Lets use memcpy to avoid a for loop
+					unsafe {
 						for (int i = 0; i < connectionParams.NumMaps; i++) {
-							fixed (byte* arrAddr = &Maps[i].Value [0])
-								Runtime.memcpy (bufferEnd, (IntPtr) arrAddr, 128);
+							fixed (void* arrAddr = &Maps[i].Value [0])
+								Buffer.MemoryCopy ((void*) bufferEnd, arrAddr, 128, 128);
```

Comment:
Buffer.MemoryCopy has the source/destination parameters reversed wrt memcpy, but they're the same order here.

A simpler change would have been to implement `Runtime.memcpy` using `Buffer.MemoryCopy` :smile:

Reply:
heh, I wanted the name to disappear :)